app/src/main/java/org/oppia/android/app/profileprogress/ProfileProgressHeaderViewModel.kt

```diff
@@ -42,7 +48,9 @@ class ProfileProgressHeaderViewModel(activity: AppCompatActivity, fragment: Frag
   }
 
   fun clickOnViewAll() {
-    routeToRecentlyPlayedActivity.routeToRecentlyPlayed()
+    routeToRecentlyPlayedActivity.routeToRecentlyPlayed(
```

Comment:
@BenHenning here, I want to compute the title as we have done in `PromotedStorylistViewModel`, is there any way we can get appropriate title for RecentlyPlayedActivity without injecting PromotedStoryListViewModel, because if we inject PromotedStoryListViweModel it will create issues in other files are dependent on this.

Reply:
As discussed over chat, since we're changing the behavior in the profile progress screen I think the easiest way to start is just to duplicate the existing set up from the homescreen. Once that's done, we can consider how de-duplication may be possible (but it may not reasonably be).

Reply:
I have updated this based on the logic that if we have more than 2 stories completed then the title should be `Recently Played Stories` or else it will be `Stories For You`. I have written this logic based on the condition of disabling View-All button mentioned in `PromotedStoryListViewModel`. I have also verified this manually and it's working fine.

Reply:
We need to make sure that there's an automated test for that, as well.

Regarding the logic, what exactly does the 'View All' button have to do with the header logic? It seems to me that the profile progress screen always contains in-progress topics only, or am I missing something?

Reply:
We already have automated tests for same in `ProfileProgressFragmentTest` which I have mentioned below